pandas/_libs/groupby_helper.pxi.in

```diff
@@ -906,15 +906,17 @@ def group_cumsum(numeric[:, :] out,
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def group_shift_indexer(int64_t[:] out, int64_t[:] labels,
-                        int ngroups, int periods):
+def group_shift_indexer(ndarray[int64_t] out, ndarray[int64_t] labels,
+                        int64_t ngroups, **kwargs):
     cdef:
         Py_ssize_t N, i, j, ii
-        int offset, sign
+        int offset, sign, periods
         int64_t lab, idxer, idxer_slot
         int64_t[:] label_seen = np.zeros(ngroups, dtype=np.int64)
         int64_t[:, :] label_indexer
 
+    periods = kwargs['periods']
```

Comment:
hmm, if you are not doing a ``.get()`` here, is there a reason you are not simply haveing ``periods=None`` in the signature? (if its optional)?